pandas/core/common.py

```diff
@@ -405,8 +407,16 @@ def random_state(state=None):
     Returns
     -------
     np.random.RandomState
+
+    ..versionchanged:: 1.1.0
+        array-like and BitGenerator (for NumPy>=1.17) object now passed to
+        np.random.RandomState() as seed
     """
-    if is_integer(state):
+    if (
+        is_integer(state)
+        or is_array_like(state)
```

Comment:
`np.random.RandomState([1, 2.5, 3])` will throw `TypeError: Cannot cast array from dtype('float64') to dtype('int64') according to the rule 'safe'`. So I think just `is_array_like` won't be enough.

Reply:
Tht's probably OK though. The traceback will point to the relevant section in NumPy, and if we want to improve that error message we can improve it in NumPy.